File: CCLib/src/java/lang/native/Thread.lua
natives["java.lang.Thread"] = natives["java.lang.Thread"] or {}

natives["java.lang.Thread"]["yield()V"] = function()
    coroutine.yield()
end

natives["java.lang.Thread"]["start0()V"] = function(this)
    createThread(this)
end

natives["java.lang.Thread"]["currentThread()Ljava/lang/Thread;"] = function()
    return getCurrentThread()
end

natives["java.lang.Thread"]["sleep(J)V"] = function(ms)
    sleep(bigintToDouble(ms))
end

File: CCLib/src/java/lang/Thread.java
package java.lang;

public class Thread implements Runnable {
    public static native void yield();
    public static native Thread currentThread();
    public static native void sleep(long ms);

    private Runnable target;

    public Thread() { }

    public Thread(Runnable target) {
        this.target = target;
    }

    public void start() {
        start0();
    }

    @Override
    public void run() {
        if(target != null) {
            target.run();
        }
    }

    private native void start0();
}